packages/create-llama/templates/types/streaming/nextjs/next.config.o11y.js

```diff
@@ -0,0 +1,27 @@
+/** @type {import('next').NextConfig} */
```

Comment:
It's already a bit of a miss with the different next.config files (I wanted to get rid of them) 

I see you added the following two changes:

**C1**:
```
config.module.rules.push({
      test: /\.node$/,
      loader: "node-loader",
    });
```

**C2:**
```   
    if (isServer) {
      config.ignoreWarnings = [{ module: /opentelemetry/ }];
    }
```

Can we leave out C2? Having some warnings sounds acceptable. 
What's the impact of C1?

Reply:
Unfortunately, we need both of them on Next 14 otherwise the code won't build. This has started happening with their new app router and I'll need to work with them on resolving it (actually related to OpenTelemetry, not us specifically).
See https://github.com/open-telemetry/opentelemetry-js/issues/4173

Reply:
What's the impact if we add both to the normal next.config.js?

Reply:
I took a look into generating this JS file. We could, of course, parse the AST, modify it, and write it to the file system.

The problem is that it makes the code hard to read/maintain.

How about we generally use the `next.config.json` below and then generate `nextConfig` and `webpackConfig` files accordingly?

```javascript
/** @type {import('next').NextConfig} */
import _ from "lodash";

// TODO: Generate nextConfig and webpackConfig files from create-llama
// import nextConfig from "./next.config.json";
const nextConfig = {
  experimental: {
    outputFileTracingIncludes: {
      "/*": ["./cache/**/*"],
    },
  },
};

// import webpackConfig from "./webpack.config.json";
const webpackConfig = {
  // See https://webpack.js.org/configuration/resolve/#resolvealias
  resolve: {
    alias: {
      sharp$: false,
      "onnxruntime-node$": false,
    },
  },
};

nextConfig.webpack = (config) => {
  return _.merge(config, webpackConfig);
};

module.exports = nextConfig;
```

Reply:
We can maybe just add them to the normal next config, or I can do that refactor - what do you prefer?

Reply:
I just merged a change to `main` that is using this pattern, please have a look at: https://github.com/run-llama/LlamaIndexTS/pull/631 - hope it works for you too!

Reply:
@marcusschiesser cool! I can add one of the 2 things that I need, I'm just left with this:
```
if (isServer) {
      config.ignoreWarnings = [{ module: /opentelemetry/ }];
    }
```
Which can't be in JSON. Are you OK with adding it to the main `next.config.mjs`? Should I write a patch that dynamically fetches an optional `.js` file in the next config?

Reply:
@nirga I see, it gets a bit tricky. I just created a new PR (https://github.com/run-llama/LlamaIndexTS/pull/634) that is using JSON for the next config and a function for webpack. Then you copy a new webpack function in your code